common/data_refinery_common/job_lookup.py

```diff
@@ -23,13 +34,129 @@ class ProcessorPipeline(PipelineEnums):
+        # Sometimes Array Express lies about what a sample's platform
+        # is. Therefore, if there's a .CEL file we'll download it and
+        # determine the platform from that.
+        relations = OriginalFileSampleAssociation.objects.filter(sample=sample_object)
+        original_files = OriginalFile.objects.filter(id__in=relations.values('original_file_id'))
+        for original_file in original_files:
+            if original_file.source_filename[-4:].upper() == ".CEL":
+                return Downloaders[sample_object.source_database]
+
+    return Downloaders.NONE
+
+
+def determine_processor_pipeline(sample_object: Sample) -> ProcessorPipeline:
+    """Determines the appropriate processor pipeline for the sample.
+
+    This is mostly a giant set of nested if statements, so describing
+    the logic wouldn't add very much. However, the general flow is:
+      - Is the platform supported? If not return NONE cause we don't want it.
+      - Does it have raw data? If not NO_OP the data.
+        (With one exception explained in comments.)
+      - Is it Microarray data? If so determine which processor based on its
+        manufacturer Otherwise it's SALMON-time.
+    """
+    if not _is_platform_supported(sample_object.platform_accession_code):
+        return ProcessorPipeline.NONE
+
+    if sample_object.technology == "MICROARRAY":
+        if not sample_object.has_raw:
+            return ProcessorPipeline.NO_OP
+        else:
+            if sample_object.manufacturer == "ILLUMINA":
+                return ProcessorPipeline.ILLUMINA_TO_PCL
+            elif sample_object.manufacturer == "AFFYMETRIX":
+                return ProcessorPipeline.AFFY_TO_PCL
+            elif sample_object.manufacturer == "AGILENT":
+                # We currently aren't prepared to process Agilent because we don't have
+                # whitelist of supported platforms for it. However this code works so
+                # let's keep it around until we're ready for Agilent.
+                annotations = sample_object.sampleannotation_set.all()[0]
+                channel1_protocol = annotations.data.get('label_protocol_ch1', "").upper()
+                channel2_protocol = annotations.data.get('label_protocol_ch2', "").upper()
+                if ('AGILENT' in channel1_protocol) and ('AGILENT' in channel2_protocol):
+                    return ProcessorPipeline.AGILENT_TWOCOLOR_TO_PCL
+                else:
+                    return ProcessorPipeline.AGILENT_ONECOLOR_TO_PCL
+            elif sample_object.manufacturer == "UNKNOWN":
+                logger.error("Found a Sample on a supported platform with an unknown manufacturer.",
+                             sample=sample_object.id,
+                             platform_accession=sample_object.platform_accession_code)
+                return ProcessorPipeline.NONE
+
+    else:
+        if not sample_object.has_raw:
+            # Only NO_OP RNASeq data if it's from SRA, because
```

Comment:
To my knowledge, SRA doesn't contain any processed data. Have you found an example of processed data there? I don't think it will matter because of the way you have the logic here, but we wouldn't want to NOOP anything from SRA if there's no processed data (unless there's a metadata consideration here I am unaware of).

Reply:
Ok, so I'll change the RNA-SEQ logic block to just check if there is raw data and if so to SALMON, otherwise don't do anything.